encord/storage.py

```diff
@@ -53,17 +55,44 @@ def parent(self) -> Optional["StorageFolder"]:
     def name(self) -> str:
         return self._orm_folder.name
 
+    @name.setter
+    def name(self, name: str):
```

Comment:
I'm not a huge fan of property setters, that do something like that. They are too opaque for users, and generally sets the habit of modifying our library objects using assignment, that not always work great. 
So I would lean towards explicit setter methods in one or another

Reply:
Makes sense.

What's your preference between

```
item.set_name(new_name)
item.set_client_metadata(new_metadata)
```

vs

```
item.update(
    name=new_name,
    description=None, # don't change
    client_metadata=new_metadata,
)
```

The first one is better self-documentation but the second one is better for batch updates as it's a single BE call.

Or maybe both?